### PR TITLE
UW-271 Implement -q/-v args as mutually exclusive

### PR DESCRIPTION
**Pull request by MYWeidner**
**Description**
Made -q and -v mutually exclusive in set_config.py and templater.py. Excluded dry run from this change as dry run with -q or -v flags is appropriate. Created tests for this change in test_set_config.py and test_templater.py. No changes were made to at parse_to_jinja2.py as -q and -v flags are not available in this file.

Completed for issue [162](https://github.com/ufs-community/workflow-tools/issues/162#issue-1601622865) 

**Type of change**

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: tests/test_templater.py [hunk 1]
```diff
@@ -157,8 +157,7 @@ def test_set_template_no_config_suffix_fails(): #pylint: disable=unused-variable
         args = [
             '-i', input_file,
             '-c', tmp_file.name,
-            '-d',
-            '-q',
+            '-d'
```
Comment (christinaholtNOAA) [inline tests/test_templater.py:160]: Super tiny suggestion here -- a Python life hack, really. When managing lists over multiple lines, it's often helpful to include the comma after the last item for ease of supportability. If someone (or some tool) came in to rearrange the order or the list, or add another item to the list, there'd be no need to remember to add the extra comma here. Things move around in future iterations much more fluidly with that pesky extra comma.

File: tests/test_templater.py [hunk 2]
```diff
@@ -327,3 +325,13 @@ def test_set_template_verbosity(): #pylint: disable=unused-variable
     outcome = ''
     # Check that only the correct messages were logged
     assert result == outcome
+
+def test_mutually_exclusive_args(): #pylint: disable=unused-variable
+    ''' Test that -q, -v, -d args are mutually exclusive'''
```
Comment (christinaholtNOAA) [inline tests/test_templater.py:330]: Here, you're suggesting that `-q`, `-v`, and `-d` should be mutually exclusive, but you only included `-q` and `-v` here in the test. They were also the only two included in the `group` addition in the `scripts` files.

Should the comment just include `-q` and `-v`?

**Review by christinaholtNOAA — COMMENTED** [review]
Mayra, this is a really nice (and quick) first addition! I have a couple of small suggestions for changes below, otherwise it looks great.

**Comment by WeirAE** [comment]
Well, dry_run was just in my original comment, but since it only limits the file not any output, probably just remove it from the comment

**Comment by christinaholtNOAA** [comment]
Do either of you happen to know if you get output if you choose `--dry-run` and `-q`? If so, leaving it as-is should be fine.

I think `--dry-run` should be able to print to screen/log no matter what, so we might consider making those mutually exclusive if the user might get baffled about something like "I added the dry-run flag but it isn't telling me anything".

**Comment by WeirAE** [comment]
AFAIK, overlapping exclusive groups is not especially pleasant. Could add a message "Quiet dry-run: this is the last message you will see - are you sure this is what you want to do?" in the meantime?

File: tests/test_templater.py (hunk 1 above)
Comment (MYWeidner) [reply tests/test_templater.py:160]: Good to know. I didn't realize we could have a comma at the end of a list.

File: tests/test_templater.py (hunk 2 above)
Comment (MYWeidner) [reply tests/test_templater.py:330]: Comment updated in test. -q and -v should be mutually exclusive.

**Comment by venitahagerty** [comment]
I don't know if I'm being confused by multiple edits, but it looks like -d is still being included in the "mutually exclusive" comments.

**Comment by MYWeidner** [comment]
I added an error for when users use -d and -q, but now I'm failing tests/test_templater.py::test_set_template_verbosity - TypeError: expected str, bytes or os.PathLike object, not NoneType. I can't figure out how to test -q without dry run flag. Does anyone have any ideas?

**Pull request merged**